src/ucp/api/ucp.h

```diff
@@ -939,20 +947,36 @@ typedef struct ucp_generic_dt_ops {
  * @ingroup UCP_DATATYPE
  * @brief UCP datatype attributes
  *
- * This structure provides attributes that can be queried for a UCP datatype.
+ * This structure provides attributes of a UCP datatype.
  */
 typedef struct ucp_datatype_attr {
     /**
      * Mask of valid fields in this structure, using bits from @ref
      * ucp_datatype_attr_field. Fields not specified in this mask will be
      * ignored. Provides ABI compatibility with respect to adding new fields.
      */
-    uint64_t field_mask;
+    uint64_t   field_mask;
+
+    /**
+     * Packed size of the given datatype. (output parameter)
+     */
+    size_t     packed_size;
 
     /**
-     * Packed size of a single element of the given datatype.
+     * Pointer to a data buffer of the associated data type.
+     * This field is optional.
+     * If @ref UCP_DATATYPE_ATTR_FIELD_BUFFER is not set in @ref field_mask,
+     * this field defaults to @e NULL.
      */
-    size_t   packed_size;
+    const void *buffer;
+
+    /**
+     * Number of elements in @a buffer.
+     * This value is optional.
+     * If @ref UCP_DATATYPE_ATTR_FIELD_COUNT is not set in @ref field_mask, the
+     * value of this field defaults to 0.
```

Comment:
maybe the default should be 1?

Reply:
I am not sure. 0 should basically signal that this value is not set (invalid). 0 would also enable zero-initialization of all fields after `packed_size`.
Since you bring this up, what do we expect the `packed_size` to be for a contiguous datatype, when `count` >1 is passed? count * _sizeOfDatatype_?

Reply:
IMO the default should be "1" and for contig type we should multiply it by sizeOfDatatype
@shamisp WDYT?